### PR TITLE
V2: Lint

### PR DESCRIPTION
**Pull request by RedHatter**
Various fixes for linting including almost all of @jscad/modeling (if someone could take a look at `modeling/src/math/OrthoNormalBasis.js`) . Should fix any issues like #569 and make future ones easier to catch as lint can now be used cleanly.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
@RedHatter this is really awesome. that's a lot of work. thanks.

Can you X those boxes in the initial comment? We just want to make sure that everyone does their part in keeping the code and the tests current.

**Comment by z3dev** [comment]
@RedHatter please keep the scope to modeling, as there are still some V2 changes required for the other modules. and as you mentioned above, we can now run lint going forward and fix issues.

File: packages/modeling/src/operations/extrusions/extrudeFromSlices.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const extrudeFromSlices = (options, base) => {
     isCapped: true,
     callback: defaultCallback
   }
-  let { numberOfSlices, isCapped, callback } = Object.assign({ }, defaults, options)
+  const { numberOfSlices, isCapped, callback: generate } = Object.assign({ }, defaults, options)
```
Comment (z3dev) [inline packages/modeling/src/operations/extrusions/extrudeFromSlices.js:47]: Why the code change? Did this pass tests?

File: packages/cli/src/determineOutputNameAndFormat.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const determineOutputNameAndFormat = (outputFormat, outputFile, inputFile) => {
     if (!outputFile) { // unless output filename not set, compose it
       ext = ext.replace(/stl[ab]/, 'stl') // drop [ab] from stl
       outputFile = inputFile
-      outputFile = outputFile.replace(/\.([^\.]+)$/, '.' + ext) // compose output filename
+      outputFile = outputFile.replace(/\.([^.]+)$/, '.' + ext) // compose output filename
```
[the remark this answers is not in the document]
Comment (RedHatter) [reply packages/cli/src/determineOutputNameAndFormat.js:17]: It's not a bug just a needless escape. `/[\.]/` and `/[.]/` are equivalent.

File: packages/io/svg-deserializer/shapesMapGeometry.js [hunk 3]
```diff
@@ -69,15 +69,15 @@ const shapesMapGeometry = (obj, objectify, params) => {
...
-      let r = cssPxUnit // default
-      if ('strokeWidth' in obj) {
-        r = cagLengthP(obj.strokeWidth, svgUnitsPmm, svgUnitsV) / 2
-      } else {
-        const v = groupValue(svgGroups, 'strokeWidth')
-        if (v !== null) {
-          r = cagLengthP(v, svgUnitsPmm, svgUnitsV) / 2
-        }
-      }
+      // let r = cssPxUnit // default
```
[the remark this answers is not in the document]
Comment (RedHatter) [reply packages/io/svg-deserializer/shapesMapGeometry.js:72]: `r` is never used. I would remove it but was unsure if it was WIP and will be used later.

File: packages/io/svg-deserializer/shapesMapJscad.js [hunk 4]
```diff
@@ -118,18 +118,18 @@ const shapesMap = function (obj, codify, params) {
...
-    r = cagLengthP(obj.strokeWidth, svgUnitsPmm, svgUnitsV) / 2
-  } else {
-    const v = groupValue(svgGroups, 'strokeWidth')
-    if (v !== null) {
-      r = cagLengthP(v, svgUnitsPmm, svgUnitsV) / 2
-    }
-  }
-  return r
-}
+// const getStrokeWidth = (obj, svgUnitsPmm, svgUnitsV, svgGroups) => {
```
[the remark this answers is not in the document]
Comment (RedHatter) [reply packages/io/svg-deserializer/shapesMapJscad.js:121]: Again, I was unsure if it was WIP and will be used later but if not I'll remove it.

File: packages/modeling/src/operations/extrusions/extrudeFromSlices.js (hunk 1 above)
Comment (RedHatter) [reply packages/modeling/src/operations/extrusions/extrudeFromSlices.js:47]: It did pass test. Nothing changed to the callee just a local variable name change. In standard js `callback` and `cb` expect an error argument see [here](https://eslint.org/docs/rules/handle-callback-err).

File: package.json [hunk 5]
```diff
@@ -45,6 +45,15 @@
     "logo": "https://opencollective.com/openjscad/logo.txt"
   },
   "standard": {
-    "ignore": [ "**/dist/", "**/old/" ]
+    "ignore": [
+      "**/dist/",
+      "**/old/",
+      "**/examples/",
+      "**/io/scad-deserializer"
+    ],
```
[the remark this answers is not in the document]
Comment (RedHatter) [reply package.json:53]: I can though `gcode-deserializer` isn't throwing any linting errors.

**Comment by RedHatter** [comment]
`ui/csg-viewer` looks to just be an older version of `utils/regl-renderer`. Should we add it to ignore list? Or maybe remove it entirely?

**Comment by RedHatter** [comment]
> @RedHatter please keep the scope to modeling (at least for this PR), as there are still some V2 changes required for the other modules

I know there are many changes still being worked on but I see no reason not to fix the linting errors currently on V2

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (z3dev) [reply package.json:53]: Let's take out. It's basically the same state as the scad translator.

File: packages/modeling/src/operations/extrusions/extrudeFromSlices.js (hunk 1 above)
Comment (z3dev) [reply packages/modeling/src/operations/extrusions/extrudeFromSlices.js:47]: That link seems to be rules in regards to proper error handling in callback functions, i.e. promises.

Was eslint throwing warnings for the 'callback' attribrute name?

Comment (RedHatter) [reply packages/modeling/src/operations/extrusions/extrudeFromSlices.js:47]: It's for node style callbacks, nothing to do with Promises.

Yes, if it's called `callback` or `cb` it is assumed to be a node style callback.

File: packages/web/src/utils/urlUtils.js [hunk 6]
```diff
@@ -1,14 +1,13 @@
 const url = require('url')
 
 const fetchUriParams = (uri, paramName, defaultValue = undefined) => {
-  const params = url.parse(uri, true)
```
Comment (z3dev) [inline packages/web/src/utils/urlUtils.js]: This doesn't seem like a change based on lint. It's probablly better to put small fixes to logic into another PR, along with a few simple test cases.

File: packages/modeling/src/operations/extrusions/extrudeFromSlices.js (hunk 1 above)
Comment (z3dev) [reply packages/modeling/src/operations/extrusions/extrudeFromSlices.js:47]: I see... OK. I'm fine with this change then.

File: packages/web/src/utils/urlUtils.js (hunk 6 above)
Comment (RedHatter) [reply packages/web/src/utils/urlUtils.js]: Lint complains about `url.parse` being deprecated.

Comment (RedHatter) [reply packages/web/src/utils/urlUtils.js]: And it should be functionally equivalent.

File: packages/io/svg-deserializer/shapesMapJscad.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (z3dev) [reply packages/io/svg-deserializer/shapesMapJscad.js:121]: Yeah. WIP. OK.

File: packages/io/svg-deserializer/shapesMapGeometry.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (z3dev) [reply packages/io/svg-deserializer/shapesMapGeometry.js:72]: OK. Let's comment the code for now. Someday... this may be implemented.

File: packages/web/src/utils/urlUtils.js (hunk 6 above)
Comment (z3dev) [reply packages/web/src/utils/urlUtils.js]: Can we test it? Or maybe I should say.. did you test it? :)

Comment (RedHatter) [reply packages/web/src/utils/urlUtils.js]: I ran some quick tests in console but didn't write CI test or do anything extensive.

If you're not conferable with the change I can revert it.

**Comment by RedHatter** [comment]
That's all for now. I'm not completely sure how to resolve the rest of the lint errors. Someone else will have to take a look.

File: packages/web/src/utils/urlUtils.js (hunk 6 above)
Comment (z3dev) [reply packages/web/src/utils/urlUtils.js]: @kaosat-dev thoughts? i'm not a web developer by trade. ;)

**Comment by z3dev** [comment]
@kaosat-dev please review. (Warning... it takes time to review all the changes.)

I think these changes will bring the code up to our standards, and force further changes to adhere.

For the other pul requests, we can just merge V2 again, and resubmit.

**Comment by z3dev** [comment]
> `ui/csg-viewer` looks to just be an older version of `utils/regl-renderer`. Should we add it to ignore list? Or maybe remove it entirely?

This is being used by the 'desktop' application, so it probably needs to stay around for now.

File: packages/io/io/formats.js [hunk 7]
```diff
@@ -111,95 +122,95 @@ const conversionFormats = [
...
-  },
-  {
-    extensions: ['woff2'],
-    type: 'font',
-    binary: false,
-    convertableToJscad: false
-  }
-
-]
+// const supportedInputFormats = [
```
[the remark this answers is not in the document]
Comment (z3dev) [reply packages/io/io/formats.js:125]: @RedHatter please drop this function. Remove the code.

File: packages/web/src/utils/urlUtils.js (hunk 6 above)
Comment (z3dev) [reply packages/web/src/utils/urlUtils.js]: @RedHatter please revert. We can address later.

**Comment by RedHatter** [comment]
@z3dev Done

@kaosat-dev Should be ready to merge

**Comment by z3dev** [comment]
@RedHatter please merge the lastest V2 into this set of changes. there may be a few conflicts with doc strings, as the lastest merge contained changes for JSDOC generation.

**Comment by z3dev** [comment]
@RedHatter please resolve the 'Conflicting files', and then we can merge.

**Comment by RedHatter** [comment]
Rebased off of V2.

**Comment by kaosat-dev** [comment]
thanks @RedHatter ! woohoo !
@z3dev I will merge this tomorrow morning, I need to double check one last thing before it can be done, any objections on your end ?

**Comment by z3dev** [comment]
> @z3dev I will merge this tomorrow morning, I need to double check one last thing before it can be done, any objections on your end ?

Reviewed the latest changes. Looks fine.
@RedHatter thanks again.

@kaosat-dev this is ready to merge.
